src/main/java/io/strimzi/kafka/quotas/StaticQuotaCallback.java

```diff
@@ -185,17 +196,49 @@ private void closeExecutorService() {
         }
     }
 
+    static MetricName metricName(Class<?> clazz, String name, LinkedHashMap<String, String> tags) {
+        String group = clazz.getPackageName();
+        String type = clazz.getSimpleName();
+        return metricName(name, type, group, tags);
+    }
+
     static MetricName metricName(Class<?> clazz, String name) {
         String group = clazz.getPackageName();
         String type = clazz.getSimpleName();
         return metricName(name, type, group);
     }
 
-    static MetricName metricName(String name, String type, String group) {
+    /**
+     * Generate an mBean metric name
+     * @param name  the name of the Metric.
+     * @param type  the type to which the Metric belongs.
+     * @param group the group to which the Metric belongs type
+     * @return the MetricName object derived from the arguments.
+     */
+    public static MetricName metricName(String name, String type, String group) {
         String mBeanName = String.format("%s:type=%s,name=%s", group, type, name);
         return new MetricName(group, type, name, SCOPE, mBeanName);
```

Comment:
Does Yammer's `MetricName` constructor validate the `mBeanName` wrt the rules defined in [javax.management.ObjectName](https://docs.oracle.com/javase/8/docs/api/javax/management/ObjectName.html)? If not should we do that here, or are we content that the only callsites in this project can never contain any illegal characters?

Reply:
Hmm Yammer is not validating. 

Have sanitised the immediate issues in 971b015f44c06a7ff25c17db6a39cb39e5fda0e1

However I want to revisit and check on quoting rules etc.

Reply:
Further tweaking in 50ddb4f21867de437eeecf80e4f1e23dd6175973 & 4cbd9549e64463c323d164d43b9a6991d94fc19b. 

I've taken the slightly draconian view that `*` `?` are totally banned rather than trying to validate they are properly escaped as I think they are likely to cause trouble in downstream metrics systems even if they are technically valid in an ObjectName.